### PR TITLE
Mnes 1189 discovery list for multiple identical aas ids

### PR DESCRIPTION
**Pull request by FranzXitaso**
When there is an AAS with the same AAS ID in multiple repos, after searching, we want to display a result list of all encountered AAS with the same ID, and then let the user select which AAS to display in the detail view.

### CODE REVIEW

File: src/lib/i18n/de.mnestix.ts [hunk 1]
```diff
@@ -245,10 +245,10 @@ export const deMnestix = {
         logoHelperText: 'base64 encoded',
     },
     discoveryList: {
-        header: 'Ergebnisliste der Discovery',
+        header: 'Zugehörige AAS IDs für',
         aasIdHeading: 'AAS ID',
         repositoryUrl: 'Repository Url',
-        subtitle: 'Asset ID',
+        subtitle: 'Sie finden alle zugehörigen AASs hier.',
```
Comment (JonathanXITASO) [inline src/lib/i18n/de.mnestix.ts:251]:
```suggestion
        subtitle: 'Sie finden alle zugehörigen Verwaltungsschalen hier.',
```

File: src/lib/i18n/en.mnestix.ts [hunk 2]
```diff
@@ -244,10 +244,10 @@ export const enMnestix = {
         logoHelperText: 'base64 encoded',
     },
     discoveryList: {
-        header: 'Discovery result list',
+        header: 'Related AAS IDs to',
         aasIdHeading: 'AAS ID',
         repositoryUrl: 'Repository Url',
-        subtitle: 'Asset ID',
+        subtitle: 'You will find all related AASs from here.',
```
Comment (JonathanXITASO) [inline src/lib/i18n/en.mnestix.ts:250]:
```suggestion
        subtitle: 'You will find all related administration shells from here.',
```

File: src/lib/i18n/en.mnestix.ts [hunk 3]
```diff
@@ -244,10 +244,10 @@ export const enMnestix = {
         logoHelperText: 'base64 encoded',
     },
     discoveryList: {
-        header: 'Discovery result list',
+        header: 'Related AAS IDs to',
```
Comment (JonathanXITASO) [inline src/lib/i18n/en.mnestix.ts:247]: From OpenHour:
```suggestion
        header: 'Related AAS to',
```

File: src/lib/i18n/de.mnestix.ts [hunk 4]
```diff
@@ -245,10 +245,10 @@ export const deMnestix = {
         logoHelperText: 'base64 encoded',
     },
     discoveryList: {
-        header: 'Ergebnisliste der Discovery',
+        header: 'Zugehörige AAS IDs für',
```
Comment (JonathanXITASO) [inline src/lib/i18n/de.mnestix.ts:248]: From OpenHour
```suggestion
        header: 'Zugehörige Verwaltungsschalen für',
```

File: src/lib/searchUtilActions/SearchRepositoryHelper.ts [hunk 5]
```diff
@@ -32,11 +48,18 @@ export async function getAasThumbnailFromAllRepos(
...
+            if (image.size === 0) {
+                throw new Error('Empty image');
+            }
             return image;
-        }
+        }),
+    );
+
+    try {
+        return await Promise.any(promises);
```
Comment (JonathanXITASO) [inline src/lib/searchUtilActions/SearchRepositoryHelper.ts:61]: Just to make sure:
 This code will only throw, if ALL `promises` throw an error?
 Else we want to take the first valid one, i think.

**Review by JonathanXITASO — CHANGES_REQUESTED** [review]

File: src/app/[locale]/viewer/discovery/_components/DiscoveryListView.tsx [hunk 6]
```diff
@@ -21,38 +25,57 @@ export const DiscoveryListView = () => {
...
-                    const aasRepositoryUrl = registrySearchResult
-                        ? registrySearchResult.registryAasData?.aasRegistryRepositoryOrigin
-                        : (await isAasAvailableInRepo(aasId, env.AAS_REPO_API_URL))
-                          ? env.AAS_REPO_API_URL
-                          : '-';
-                    entryList.push({
-                        aasId: aasId,
-                        repositoryUrl: aasRepositoryUrl,
-                    });
+            // TODO do it asynchronous
```
[the remark this answers is not in the document]
Comment (Xmilofranke) [reply src/app/[locale]/viewer/discovery/_components/DiscoveryListView.tsx]: I solved the todo in this pr

File: src/app/[locale]/viewer/discovery/_components/DiscoveryListView.tsx [hunk 7]
```diff
@@ -37,8 +38,7 @@ export const DiscoveryListView = () => {
         if (assetId) {
             const aasIds = await handleAasDiscoverySearch(assetId);
 
-            // TODO do it asynchronous
-            for (const aasId of aasIds) {
+            aasIds.map(async (aasId) => {
```
Comment (JonathanXITASO) [inline src/app/[locale]/viewer/discovery/_components/DiscoveryListView.tsx:41]: do you need to await the results of this map in a `Promise.all()`?

**Review by JonathanXITASO — APPROVED** [review]

**Pull request merged**